src/visitor.rs

```diff
@@ -47,6 +48,30 @@ fn is_extern_crate(item: &ast::Item) -> bool {
     }
 }
 
+/// Creates a string slice corresponding to the specified span.
+pub struct SnippetProvider<'a> {
+    /// A pointer to the content of the file we are formatting.
+    big_snippet: &'a Rc<String>,
```

Comment:
Any reason for not cloning `Rc` (and `big_snippet: Rc<String>`)? It is just an increment, and you can get rid of a layer of indirection ("`&` -> `Rc` -> `String`" to "`Rc` -> `String`"), which might be good for performance.